automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala

```diff
@@ -419,5 +419,13 @@ class NotebookInteractionSpec extends FreeSpec with LeonardoTestUtils with Befor
         }
       }
     }
+
+    "should throw 401 for invalid token" in withWebDriver { implicit driver =>
+      try {
+        Leonardo.notebooks.setCookie(ronCluster.googleProject, ronCluster.clusterName)
+      } catch {
+        case e: Exception => e.getMessage should include(""""statusCode":409""")
+      }
```

Comment:
So here you're correctly expecting the `setCookie` call to fail, and verifying the error message. But if the call to `/setCookie` unexpectedly _succeeds_, then the test will pass. A pattern for negative testing is often the following:

```
try {
  // call thing expected to fail
  fail("Expected thing to fail, but got success")
} catch {
  case e: Exception => // verify exception
}
```
ScalaTest actually provides a different API for doing this. Here is an example:
https://github.com/DataBiosphere/leonardo/blob/develop/automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/NotebookInteractionSpec.scala#L113-L119

I would recommend doing something like the above link.